test/Infer/pruning.opt

```diff
@@ -0,0 +1,27 @@
+; REQUIRES: solver, synthesis
+; RUN: %souper-check -infer-rhs -souper-exhaustive-synthesis -souper-dataflow-pruning -souper-exhaustive-synthesis-debug-level=2 %solver %s > %t1
+; RUN: %FileCheck %s < %t1
+
+%0:i32 = var
+%1 = add %0, 1
+%2 = and %1, 5
+infer %2
+; Dataflow Pruned
+; CHECK: %0:i32 = var ; 0
+; CHECK: %1:i32 = var ; reservedconst_0
```

Comment:
ok, so I'm making sure I understand this: this is an entire RHS (except missing the synthesized constant) and you've proved that we don't need to try to synthesize a constant at al?

Reply:
also I think you want CHECK-NEXT for these lines that are consecutive in the output-- reduces odds of spurious matches

Reply:
CHECK-NEXT doesn't seem to work because:
a
b
a
c

;CHECK: a
;CHECK-NEXT: c

Fails because it finds a \n b